### PR TITLE
feat: collection page sidebar data

### PR DESCRIPTION
**Pull request by kendallstrautman**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-kscollection-page-skeleton-hashicorp.vercel.app/vault/tutorials/adp) 🔎
- [Asana task](url) 🎟️

## What

This PR adds sidebar sort order functionality to the `getAllCollections` call, which is used for the collection and tutorial product page views. 

It also adds a basic list of the sorted sidebar items for the collection view. 

## Why

To wire up this data properly for UI consumption.

## Testing

- [Visit this page](https://dev-portal-git-kscollection-page-skeleton-hashicorp.vercel.app/vault/tutorials/adp) and look at the list of the sidebar items. It should match the [Learn version](https://learn.hashicorp.com/collections/vault/getting-started) today (even though categories aren't listed yet).

Screen Shot 2022-04-07 at 3 01 12 PM

- [Same with Waypoint](https://dev-portal-git-kscollection-page-skeleton-hashicorp.vercel.app/waypoint/tutorials/deploy-azure)

Screen Shot 2022-04-07 at 3 03 07 PM

## Anything else?

Next steps are incorporating the proper sidebar / sidecar layout!

### CODE REVIEW

File: src/lib/learn-client/api/collection/fetch-product-collections.ts [hunk 1]
```diff
@@ -11,11 +16,25 @@ export const PRODUCT_COLLECTION_API_ROUTE = (
  * Collections will be product associated if they have at least
  * one tutorial whose primary product (first in frontmatter array)
  * matches this product slug
+ *
+ * Optional query params will return collections filtered by
+ * theme or sorted according to the category sidebar sort
```
Comment (kendallstrautman) [inline src/lib/learn-client/api/collection/fetch-product-collections.ts:21]:
```suggestion
 * Optional query params will return collections 
 * sorted according to the category sidebar sort (which automatically
 * includes filtering for theme
```

File: src/lib/learn-client/api/collection/fetch-product-collections.ts [hunk 2]
```diff
@@ -11,11 +16,26 @@ export const PRODUCT_COLLECTION_API_ROUTE = (
...
+  product: AllCollectionsProductOptions
 ): Promise<Collection[]> {
-  const route = PRODUCT_COLLECTION_API_ROUTE(identifier)
+  const baseUrl = PRODUCT_COLLECTION_API_ROUTE(product.slug)
+  let route = baseUrl
+
+  if (product.sidebarSort) {
+    const params = new URLSearchParams([
+      ['topLevelCategorySort', 'true'],
+      ['theme', product.slug],
```
Comment (kendallstrautman) [inline src/lib/learn-client/api/collection/fetch-product-collections.ts:33]: Turns out that the `theme` filtering _only_ works in combination with sidebar sort. Made a ticket to follow up in the API board but not a blocker for now, we can filter on the frontend.

**Comment by kendallstrautman** [comment]
Hmm I wonder why the lint check is failing it's working for me locally 🤔

**Review by zchsh — APPROVED** [review]
🎉 Looks good to me - sidebar items seem to match 👍

File: src/pages/vault/tutorials/[...tutorialSlug]/index.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { GetStaticPathsResult } from 'next'
 import vaultData from 'data/vault.json'
-import { ProductData } from 'types/products'
+import { LearnProductData } from 'types/products'
```
Comment (kendallstrautman) [inline src/pages/vault/tutorials/[...tutorialSlug]/index.tsx:3]: See changes pulled in from #278

**Pull request merged**